src/components/builder.ts

```diff
@@ -235,6 +235,9 @@ export class Builder {
         } else if (!step.isExternal) {
             if (step.command === 'latexmk' && step.rootFile === this.extension.manager.localRootFile && this.extension.manager.rootDir) {
                 cwd = this.extension.manager.rootDir
+                if (step.args && !step.args.includes('-cd')) {
+                    step.args.push('-cd')
+                }
```

Comment:
We have had a lot of troubles with the `-cd` option of `latexmk`. See for instance #1932
Why do we need to enforce `-cd` here?

Reply:
Without enforcing `-cd`, the test `build sub.tex to outdir` will fail. This test has a structure like this:
- main.tex: `\subfile{sub/s}`
- sub/s.tex: `\input{./subsub/infile`, `\documentclass[../main.tex]{subfiles}`
- sub/subsub/infile.tex

As `sub/s.tex` is a `subfile`, it should be able to compile independently. So the `input` should receive `./subsub/infile` which is a relative path to `sub/s.tex`.

However, without a `-cd`, the compilation initiated from `sub/s.tex` will fail, as the cwd is `./` instead of `./sub/`

`cwd = this.extension.manager.rootDir`

I have no idea why this test passed in the previous test implementation. But I think the current behavior and fail is reasonable.

As to #1932, I will write a quick test to see how to work around it.

Reply:
I've added a new test on #1932 with subfile and makeindex, compiled with latexmk. The test passed with `-cd`! Maybe `latexmk` has fixed the bug? https://github.com/James-Yu/LaTeX-Workshop/issues/1932#issuecomment-583258858

Reply:
I should have noted that the new test on `makeindex` with `-cd` is `build sub.tex to outdir with makeindex` in `01_build`. Will add the issue no. to the test name.